packages/state-manager/src/sagas/messages/verifyMessage/verifyMessages.saga.ts

```diff
@@ -1,14 +1,49 @@
 import { type PayloadAction } from '@reduxjs/toolkit'
-import { put } from 'typed-redux-saga'
+import { select, call, put, delay } from 'typed-redux-saga'
 import { messagesActions } from '../messages.slice'
-import { type MessageVerificationStatus } from '@quiet/types'
+import { MessageType, type MessageVerificationStatus } from '@quiet/types'
+import { publicChannelsSelectors } from '../../publicChannels/publicChannels.selectors'
+
+import { usersSelectors } from '../../users/users.selectors'
+import { verifyUserInfoMessage } from '@quiet/common'
 
 export function* verifyMessagesSaga(
   action: PayloadAction<ReturnType<typeof messagesActions.incomingMessages>>['payload']
 ): Generator {
   const messages = action.payload.messages
 
   for (const message of messages) {
+    let ownerData = yield* select(usersSelectors.ownerData)
```

Comment:
Code starting from line 16 to 24 should be moved out of the `for` loop as it doesn't depend on the message's data and makes no sense to repeat on each iteration